### PR TITLE
Basic express app added

### LINKED ISSUE #12: Setup an express app and create folder backend
**Is your feature request related to a problem? Please describe.**
setup an initial express app using the standard method and just setup the app and run , add proper environment variable for PORTS and other things the expressapp should be under ```backend``` folder under the main repository. 

name of the npm package should be : backendhotelontouch

### PR DESCRIPTION
**Pull request by chaimat**
Fixes #12 

Created a basic express app with a dynamic ENV port and a test route.

- [X]

Folders Changed -
- backend (created)

Screenshots of the changes (If any) -

Note: Please check Allow edits from maintainers. if you would like us to assist in the PR.

#### ✅️ By submitting this PR, I have verified the following

- [X] Worked on forked repo and starred the main repo
- [X] Checked to see if a similar PR has already been opened 🤔️
- [X] Reviewed the contributing guidelines 🔍️
- [X] Sample preview link added (add the link(s) for all the pages changed/updated from the checks tab after checks complete)
- [X] Tried Squashing the commits into one

### CODE REVIEW

**Review by ayan-biswas0412 — CHANGES_REQUESTED** [review]
Please integrate the mongoose framework and connect a local mongodb instance , give the database name as : hotelontouchdb and connect it to the expressapp , use proper model, controller, route , declare one SampleModel.js , SampleController.js, SampleRoute.js 

The file structure should be ```model/SampleModel.js```
```controller/SampleController.js``` 
```route/SampleRoute.js``` and about the mongoose you can refer the [documentation](https://mongoosejs.com/docs/)

File: backend/controller/SampleController.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+          if (!data){
+            return res
+              .status(404)
+              .send("Data with id not found");
+          }
+        
+          res.send(data);
+    },
+
+    async remove (req, res) {
```
Comment (ayan-biswas0412) [inline backend/controller/SampleController.js:45]: don't use async functions please convert them to normal function with the .then() and .catch()

**Review by ayan-biswas0412 — CHANGES_REQUESTED** [review]
Run a short readme like how to setup the thing locally and run the server 
Include all steps necessary for mongoDB setup

File: backend/controller/SampleController.js (hunk 1 above)
Comment (chaimat) [reply backend/controller/SampleController.js:45]: Sure, will do that

Comment (ayan-biswas0412) [reply backend/controller/SampleController.js:45]: make sure your code passes all the CI tests

**Comment by chaimat** [comment]
@ayan-biswas0412 could you please suggest what am I doing wrong that leads to the failing of the CI test, I am not able to figure it out.

**Comment by ayan-biswas0412** [comment]
> @ayan-biswas0412 could you please suggest what am I doing wrong that leads to the failing of the CI test, I am not able to figure it out.

I think it was a false positive alert from CI don't worry after reviewing the entire thing I will give you update

**Review by ayan-biswas0412 — CHANGES_REQUESTED** [review]
add a start script such that if a person performs ```npm start``` it starts the server

File: backend/README.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
+## Getting Started
+
+This README will take you through the process to setup the backend folder locally.
+
+### Prerequisites
+
+The following technologies are required to run the server and use it locally.
+
+- nodejs
```
Comment (ayan-biswas0412) [inline backend/README.md:9]: embed the links from where the following technlogies can be installed

**Review by ayan-biswas0412 — CHANGES_REQUESTED** [review]
The Readme document has no instruction how to make the database please include it
in the ```SampleRoute.js``` use a path such that each request of sample route can be accessed at ```localhost:port/sampleroute/<paths>``` , you have done it in app.js but do it in SampleRoute.js and also make sure to make a index.js file inside ```route``` and import it in app.js , make sure not to import individual import all routes and use the command app.use() multiple times for multiple paths

File: backend/README.md (hunk 2 above)
Comment (ayan-biswas0412) [reply backend/README.md]: and I think the command is mongo not ```mongod``` to start MongoDB shell is it ?

Comment (chaimat) [reply backend/README.md]: Yes ```mongo``` is used to start the mongo shell, but here we need to start a local mongodb server not the shell, and most installation procedures make you set ```mongod``` as an alias to start the mongodb server

**Comment by chaimat** [comment]
Just a clarification on the ```localhost:port/sampleroute/<paths>```
- So when the user makes a get request to ```localhost:port/sampleroute/all``` they will get all the data
- If they make a delete request to  ```localhost:port/sampleroute/delete/:id``` they will be able to delete that particular data

Please do confirm if this is what is expected and if I got it wrong then please share a few examples/ test cases of what you expect.

**Comment by ayan-biswas0412** [comment]
> Just a clarification on the `localhost:port/sampleroute/<paths>`
> 
> * So when the user makes a get request to `localhost:port/sampleroute/all` they will get all the data
> * If they make a delete request to  `localhost:port/sampleroute/delete/:id` they will be able to delete that particular data
> 
> Please do confirm if this is what is expected and if I got it wrong then please share a few examples/ test cases of what you expect.

no i am telling that soecify your path ```/sampleroute``` in the ```SampleRoute.js``` not under ```app.js``` and also make one index.js file under ```route``` which should be called in ```app.js``` which will reutrn all the routes defined under ```route``` folder => actually when we will develop another routes in the ```route``` folder we don't want to change the app.js file again and again to import routes

**Comment by ayan-biswas0412** [comment]
@chaimat are you working in this ?

**Comment by chaimat** [comment]
Yes, I am. Had an exam so didn't get much time sorry!

**Comment by ayan-biswas0412** [comment]
> Yes, I am. Had an exam so didn't get much time sorry!

ohhk ohhk no issues

**Comment by chaimat** [comment]
@ayan-biswas0412 are there any more changes to work on?

File: backend/route/app.js [hunk 3]
```diff
@@ -0,0 +1,6 @@
+const sampleroute = require('./SampleRoute')
```
Comment (ayan-biswas0412) [inline backend/route/app.js:1]: name this file as index.js and make required changes and also document the whole flow as a Development guide in this folder Readme

**Review by ayan-biswas0412 — CHANGES_REQUESTED** [review]

**Comment by ayan-biswas0412** [comment]
> @ayan-biswas0412 are there any more changes to work on?

see I have mentioned some changes

File: backend/route/app.js (hunk 3 above)
Comment (chaimat) [reply backend/route/app.js:1]: I am not familiar with the development guide part, could you link me to some examples that I could refer to?

Comment (ayan-biswas0412) [reply backend/route/app.js:1]: you can take help from here https://github.com/ayan-biswas0412/gssoc2021-HotelOnTouch/tree/main/hotelontouch
it is a guide for React App please document your thing as per your file structure

**Review by ayan-biswas0412 — APPROVED** [review]
nice work

**Pull request merged**